src/extra/stream/client.test.ts

```diff
@@ -1,22 +1,94 @@
 // @vitest-environment happy-dom
 
-import { describe, expect, test } from 'vitest';
+import { vi, beforeEach, afterEach, describe, expect, test } from 'vitest';
 import { RobotClient } from '../../robot';
 import { events } from '../../events';
+import { StreamServiceClient } from '../../gen/proto/stream/v1/stream_pb_service';
 import { StreamClient } from './client';
 
+let robotClient: RobotClient;
+let streamClient: StreamClient;
+
 describe('StreamClient', () => {
+  beforeEach(() => {
+    vi.useFakeTimers();
+
+    vi.mock('./robot/client');
+
+    robotClient = new RobotClient('fakehost');
+    vi.mock('./gen/proto/stream/v1/stream_pb_service');
+
+    streamClient = new StreamClient(robotClient);
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+    vi.useRealTimers();
+  });
+
   test('webrtc track will cause the client to emit an event', () =>
     new Promise<void>((done) => {
-      const host = 'fakeServiceHost';
-      const client = new RobotClient(host);
-      const streamClient = new StreamClient(client);
-
       streamClient.on('track', (data) => {
         expect((data as { mock: true }).mock).eq(true);
         done();
       });
 
       events.emit('track', { mock: true });
     }));
+
+  test('getStream creates and returns a new stream', async () => {
+    const fakeCamName = 'fakecam';
+    const fakeStream = { id: fakeCamName };
+    StreamServiceClient.prototype.addStream = vi
```

Comment:
Since the `StreamClient` interacts with multiple methods of the `StreamServiceClient`, partially mocking out `StreamServiceClient` by making a single one of its methods risks making these tests brittle.

I'd recommend fixing the `vi.mock` call above to point to the correct path. That way, the entire `StreamServiceClient` will be fake. From there, I think you'll be able to do something like:

```ts
StreamServiceClient
  .prototype
  .addStream
  .mockImplementation((_req, _md, cb) => {
  // ...
```

For TypeScript support, you have to wrap the fake thing in `vi.mocked`:

```ts
vi.mocked(StreamServiceClient.prototype.addStream)
  .mockImplementation((_req, _md, cb) => {
  // ...
```

Reply:
to confirm, the entire module will be mocked once i fix the mock path [as you pointed out here](https://github.com/viamrobotics/viam-typescript-sdk/pull/141/files#r1277791295), right? this was my intention

Reply:
> For TypeScript support, you have to wrap the fake thing in vi.mocked

hm, when I used `vi.mocked` as you described i got a lint exception  `@typescript-eslint/unbound method` that I wasn't quite able to figure out